ci/forms.py

```diff
@@ -165,3 +173,40 @@ class ReportGenerationForm(forms.Form):
     email_subject = forms.CharField(
         widget=forms.TextInput(attrs={'placeholder': 'Subject for email'}),
         required=False)
+
+
+class MsisdnReportGenerationForm(ReportGenerationForm):
+    msisdn_list = forms.FileField()
+
+    def clean_msisdn_list(self):
+        msisdns = []
+        wb = load_workbook(self.cleaned_data['msisdn_list'])
+        ws = wb[wb.sheetnames[0]]  # There should only be one sheet
+        # Get msisdn column
+        msisdn_column = None
+        for i, column in enumerate(ws.columns):
+            if column[0].value is None:
+                # Skip untitled columns
+                continue
+            value = str(column[0].value)
+            if value.lower() == 'phone number':
+                msisdn_column = i
+                break
+        if msisdn_column is None:
+            raise forms.ValidationError(
+                "Invalid contents for: %(file)s. File must contain "
+                "'Phone number' column",
```

Comment:
Maybe add `in the first sheet`, since we're assuming that the data is in the first sheet above.